Apps/Laboratory/src/Delegates/RestLaboratoryDelegate.cs

```diff
@@ -214,5 +223,98 @@ public async Task<RequestResult<LaboratoryReport>> GetLabReport(Guid id, string
+                this.logger.LogTrace($"Response: {response}");
+
+                switch (response.StatusCode)
+                {
+                    case HttpStatusCode.OK:
+                        this.logger.LogTrace($"Response payload: {payload}");
+                        PHSAResult<IEnumerable<PublicCovidTestResult>>? phsaResult = JsonSerializer.Deserialize<PHSAResult<IEnumerable<PublicCovidTestResult>>>(payload);
+                        if (phsaResult != null && phsaResult.Result != null)
+                        {
+                            retVal.ResultStatus = ResultType.Success;
+                            retVal.ResourcePayload = phsaResult;
+                            retVal.TotalResultCount = 1;
+                        }
+                        else
+                        {
+                            retVal.ResultError = new RequestResultError()
+                            {
+                                ResultMessage = "Error with JSON data",
+                                ErrorCode = ErrorTranslator.ServiceError(ErrorType.CommunicationExternal, ServiceType.PHSA),
+                            };
+                        }
+
+                        break;
+                    case HttpStatusCode.Forbidden:
+                        retVal.ResultError = new RequestResultError()
+                        {
+                            ResultMessage = $"DID Claim is missing or can not resolve PHN, HTTP Error {response.StatusCode}",
+                            ErrorCode = ErrorTranslator.ServiceError(ErrorType.CommunicationExternal, ServiceType.PHSA),
+                        };
+                        break;
+                    default:
+                        retVal.ResultError = new RequestResultError()
+                        {
+                            ResultMessage = $"Unable to connect to Laboratory PublicCovidTestsEndPoint, HTTP Error {response.StatusCode}",
+                            ErrorCode = ErrorTranslator.ServiceError(ErrorType.CommunicationExternal, ServiceType.PHSA),
+                        };
+                        this.logger.LogError($"Unable to connect to endpoint {endpointString}, HTTP Error {response.StatusCode}\n{payload}");
+                        break;
+                }
+            }
+#pragma warning disable CA1031 // Do not catch general exception types
+            catch (Exception e)
+#pragma warning restore CA1031 // Do not catch general exception types
+            {
+                retVal.ResultError = new RequestResultError() { ResultMessage = $"Exception getting public COVID-19 test results: {e}", ErrorCode = ErrorTranslator.ServiceError(ErrorType.CommunicationExternal, ServiceType.PHSA) };
+                this.logger.LogError($"Unexpected exception retrieving public COVID-19 test results {e}");
+            }
+            finally
+            {
+                content?.Dispose();
```

Comment:
I think content is never null.?

Reply:
Yeah, I saw that warning.  The code is borrowed from the vaccine status delegate.  I'm not sure that the warning is entirely accurate, because it seems like the variable could be null if an exception was thrown within the try block before a value was assigned to it.

Reply:
I took another and saw that this could be resolved with a using statement instead, so I've done that.